pkgs/development/tools/build-managers/tup/default.nix

```diff
@@ -0,0 +1,51 @@
+{ stdenv, fetchFromGitHub, fuse, pkgconfig }:
+
+stdenv.mkDerivation rec {
+  name = "tup-v${version}";
+  version = "0.7.3";
+
+  src = fetchFromGitHub {
+    owner = "gittup";
+    repo = "tup";
+    rev = "v${version}";
+    sha256 = "1x2grwmlf2izip4djb8cjwgl8p3x0bmfqwzjsc017mqi17qkijy8";
+  };
+
+  buildInputs = [ fuse pkgconfig ];
+
+  configurePhase = ''
+    sed -i 's/`git describe`/v${version}/g' Tupfile
+  '';
+
+  # Regular tup builds require fusermount to have suid, which nix cannot
```

Comment:
tup, here and in any other linux build of it, still includes, links, and typically makes use of fuse. By going through `tup generate`, we take advantage of the fact that none of that functionality is relevant to a one-shot build as needed for packaging. This constitutes a general solution to building packages meant to be built with tup in nix without suid fusermount, and should produce indistinguishable output up to the correctness of the implementation of `tup generate`.

The only downside is that `tup generate` seems to be poorly tested/maintained and that it forces a strictly serial build.